front/lib/swr/swr.ts

```diff
@@ -13,19 +14,57 @@ export function useSWRWithDefaults<TKey extends Key, TData>(
   fetcher: Fetcher<TData, TKey> | null,
   config?: SWRConfiguration & { disabled?: boolean }
 ) {
-  const { mutate: globalMutate } = useSWRConfig();
+  const { mutate: globalMutate, cache } = useSWRConfig();
 
   const mergedConfig = { ...DEFAULT_SWR_CONFIG, ...config };
   const disabled = !!mergedConfig.disabled;
 
   const result = useSWR(disabled ? null : key, fetcher, mergedConfig);
 
+  const mutateKeysWithSameUrl = useCallback(
+    (key: TKey) => {
+      // If the key looks like an url, we need to remove the query params
+      // to make sure we don't cache different pages together
+      // Naive way to detect url by checking for '/'
+      if (typeof key === "string" && key.includes("/")) {
+        const keyWithoutQueryParams = key.split("?")[0];
```

Comment:
Let's use the `URL` native module to achieve this.

Reply:
I wanna but the URL module expects a protocol, right ? (we get "/api/...." as keys)

Will take another look.